config/api.yml

```diff
@@ -88,6 +88,41 @@
        :post:
        - :name: create
          :identifier: alert_status_action_new
+  :alert_definition_profiles:
+    :description: Alert Definition Profiles
+    :identifier: alert_definition_profile
+    :options:
+    - :collection
+    - :subcollection
+    :verbs: *gpd
+    :klass: MiqAlertSet
+    :subcollections:
+    - :alert_definitions
+    :collection_actions:
+      :get:
+      - :name: read
+        :identifier: alert_definition_profiles_show_list
+      :post:
+      - :name: read
+        :identifier: alert_definition_profile_view
+      - :name: create
+        :identifier: alert_definition_profile_new
+      - :name: edit
+        :identifier: alert_definition_profile_edit
+      - :name: delete
+        :identifier: alert_definition_profile_delete
+    :resource_actions:
+      :get:
+      - :name: read
+        :identifier: alert_definition_profile_view
+      :post:
+      - :name: edit
+        :identifier: alert_definition_profile_edit
+      - :name: delete
+        :identifier: alert_definition_profile_delete
+      :delete:
+      - :name: delete
+        :identifier: alert_definition_profile_delete
```

Comment:
Yes. assigning miq alert to miq alert profile seems straight forward. Assigning miq alert profile to object is more complex. You can assign a miq alert profile to a specific object, all object of a provider or all object in the system ("the Enterprise"). And the UI currently has the information of which alerts can be assigned to what (based on the target type, some can be assign only to one object others only to all the objects in the system)

I don't think this would be easy to get at one shot but what we need most urgent is to assign MA to MAP and MAP to all container nodes in the system. Of course the design should allow growing other examples next